Gems/RobotecSpectatorCamera/Code/Source/SpectatorCamera/SpectatorCameraComponent.cpp

```diff
@@ -0,0 +1,258 @@
+    {
+        AZ::TickBus::Handler::BusConnect();
+        AzFramework::InputChannelEventListener::Connect();
+
+        AZ::TransformBus::EventResult(m_currentTransform, GetEntityId(), &AZ::TransformBus::Events::GetWorldTM);
+    }
+
+    void SpectatorCameraComponent::Deactivate()
+    {
+        AZ::TickBus::Handler::BusDisconnect();
+        AzFramework::InputChannelEventListener::Disconnect();
+    }
+
+    void SpectatorCameraComponent::OnTick(float deltaTime, AZ::ScriptTimePoint time)
+    {
+        if (m_configuration.m_cameraMode == CameraMode::ThirdPerson)
+        {
+            AZ::Vector3 cameraLocalPosition{ -m_orbitRadius, 0.0f, 0.0f };
+            const AZ::Quaternion yawMouseRotation = AZ::Quaternion::CreateFromAxisAngle(AZ::Vector3::CreateAxisZ(1.0f), -m_yaw);
+            const AZ::Quaternion pitchMouseRotation = AZ::Quaternion::CreateFromAxisAngle(AZ::Vector3::CreateAxisY(1.0f), m_pitch);
+            const AZ::Quaternion finalRotation = yawMouseRotation * pitchMouseRotation;
+            cameraLocalPosition = finalRotation.TransformVector(cameraLocalPosition);
+            AZ::Transform targetWorldPosition = AZ::Transform::CreateIdentity();
+            AZ::TransformBus::EventResult(targetWorldPosition, m_configuration.m_lookAtTarget, &AZ::TransformBus::Events::GetWorldTM);
+            if (m_configuration.m_followTargetRotation)
+            {
+                m_currentTransform = AZ::Transform::CreateLookAt(
+                    (targetWorldPosition.TransformPoint(cameraLocalPosition)), targetWorldPosition.GetTranslation());
+            }
+            else
+            {
+                m_currentTransform = AZ::Transform::CreateLookAt(
+                    (targetWorldPosition.GetTranslation() + cameraLocalPosition), targetWorldPosition.GetTranslation());
+            }
+        }
+        AZ::TransformBus::Event(GetEntityId(), &AZ::TransformBus::Events::SetWorldTM, m_currentTransform);
+    }
+
+    bool SpectatorCameraComponent::OnInputChannelEventFiltered(const AzFramework::InputChannel& inputChannel)
+    {
+        const AzFramework::InputDeviceId& deviceId = inputChannel.GetInputDevice().GetInputDeviceId();
+
+        if (AzFramework::InputDeviceMouse::IsMouseDevice(deviceId))
+        {
+            MouseEvent(inputChannel);
+        }
+
+        if (AzFramework::InputDeviceKeyboard::IsKeyboardDevice(deviceId))
+        {
+            KeyboardEvent(inputChannel);
+        }
+
+        return false;
+    }
+
+    void SpectatorCameraComponent::MouseEvent(const AzFramework::InputChannel& inputChannel)
+    {
+        const AzFramework::InputChannelId& channelId = inputChannel.GetInputChannelId();
+
+        if (channelId == AzFramework::InputDeviceMouse::Movement::Z && m_configuration.m_cameraMode == CameraMode::ThirdPerson)
+        {
+            if (const float newOrbitRadius =
+                    m_orbitRadius - (inputChannel.GetValue() / 1200.0f); // value 1200.0f has been chosen experimentally
```

Comment:
move the `1200.0f` to some const float variable to increase the code readability